include/tic80.h

```diff
@@ -76,14 +77,25 @@ typedef union
 {
     struct
     {
-        bool up:1;
-        bool down:1;
-        bool left:1;
-        bool right:1;
-        bool a:1;
-        bool b:1;
-        bool x:1;
-        bool y:1;
+#if RETRO_IS_BIG_ENDIAN
```

Comment:
Why is there so much of this, as long as all the code is on the same endianness, can't the compiler just sort this out itself?  Is the problem use code inside the VMs or?

Reply:
Problems arise from 2 sources:
* Loading little-endian cartridge (few places)
* Union between bit-field and plain types like u*. E.g. you can write through gamepads.first.x and read through gamepads.data. Then the result will depend on endianness unless we do those changes